### PR TITLE
allow failing tests to expand by job so you can see where they fail

### PR DESCRIPTION
**Pull request by deads2k**

### CODE REVIEW

File: pkg/api/api.go [hunk 1]
```diff
@@ -275,11 +273,11 @@ func canaryTestFailures(result map[string]v12.SortedAggregateTestsResult) []v1.C
 }
 
 // job runs with failure groups
-func failureGroupList(report v12.TestReport) []v1.FailureGroup {
+func failureGroupList(report sippyprocessingv1.TestReport) []sippyv1.FailureGroup {
```
Comment (mfojtik) [inline pkg/api/api.go:276]: @deads2k splitting these in separate commit will reduce the noise... next time :)

File: pkg/testgridanalysis/testreportconversion/testresult.go [hunk 2]
```diff
@@ -84,47 +102,64 @@ func convertRawTestResultsToProcessedTestResults(
...
 		}
-
-		filteredResults = append(filteredResults, testResult)
+		return true
 	}
+}
 
-	return filterSuccessfulTestResults(filteredResults, successThreshold)
+func filterTestResultsByFilters(fns ...testResultFilterFunc) testResultFilterFunc {
+	return func(testResult sippyprocessingv1.TestResult) bool {
```
Comment (mfojtik) [inline pkg/testgridanalysis/testreportconversion/testresult.go:133]: Stefan will love this :-)

**Comment by deads2k** [comment]
I've placed an expansion button and refactored decorations to make them more consistent.  There is more work to do on rendering.

**Comment by mfojtik** [comment]
/lgtm

Note that @deads2k won't make a career as web developer :-)

**Pull request merged**